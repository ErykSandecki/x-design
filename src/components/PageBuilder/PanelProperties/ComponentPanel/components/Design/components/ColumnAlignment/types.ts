// types
import { AlignmentHorizontal, AlignmentVertical } from 'types';
import { TButtonGroup } from 'shared/UITools/ButtonGroup/types';

export type TAligmentHorizontalButtonGroup = Array<
  Pick<TButtonGroup, 'name' | 'tooltip'> & { key: AlignmentHorizontal }
>;
export type TAligmentVerticalButtonGoup = Array<Pick<TButtonGroup, 'name' | 'tooltip'> & { key: AlignmentVertical }>;
