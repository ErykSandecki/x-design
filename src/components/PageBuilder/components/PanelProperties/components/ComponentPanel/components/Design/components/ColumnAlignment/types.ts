// types
import { AlignmentHorizontal, AlignmentVertical } from 'types';
import { TButtonGroup } from 'shared/UITools/components/ButtonGroup/types';

export type TAligmentHorizontalButtonGroup = Array<Pick<TButtonGroup, 'name'> & { key: AlignmentHorizontal }>;
export type TAligmentVerticalButtonGoup = Array<Pick<TButtonGroup, 'name'> & { key: AlignmentVertical }>;
