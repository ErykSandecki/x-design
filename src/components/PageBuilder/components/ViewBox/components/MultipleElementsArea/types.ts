// types
import { TElement } from 'types';

export type TCoordinatesData = {
  elementsCoordinates: Array<{
    coordinates: TRectCoordinates;
    id: TElement['id'];
  }>;
  outline: TRectCoordinates;
};
