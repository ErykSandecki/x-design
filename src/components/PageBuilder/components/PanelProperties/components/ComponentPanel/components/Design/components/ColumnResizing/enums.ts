export enum PopoverItem {
  auto = 1,
  fixed = 0,
  unit = 2,
}
