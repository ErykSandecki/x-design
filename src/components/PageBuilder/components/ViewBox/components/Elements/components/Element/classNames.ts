export const className = 'Element';

export const classes = {
  className,
};

export const classNames = {
  [className]: {
    name: className,
    modificators: {
      hover: `${className}--hover`,
      moving: `${className}--moving`,
      selected: `${className}--selected`,
    },
  },
};
