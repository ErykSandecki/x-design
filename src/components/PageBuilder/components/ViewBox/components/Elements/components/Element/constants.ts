// others
import { translationNameSpace as parentNameSpace } from '../../constants';

export const DATA_STATUS_ATTRIBUTE = 'data-status';
export const OUTLINE = 2;
export const THROTTLE_WAIT = 20;

export const translationNameSpace = `${parentNameSpace}.element`;
