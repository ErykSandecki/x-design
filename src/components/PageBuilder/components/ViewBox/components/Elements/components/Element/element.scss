@import '../../../../../../styles/z-indexes';
@import '../../../../../../../../styles/mixins/enter-delay';

.Element {
  $self: &;
  @include enter-delay;

  animation-delay: 100ms;
  overflow: hidden;
  position: absolute;
  user-select: none;
  z-index: $--xd-zindex-element;

  &--hover {
    z-index: $--xd-zindex-element-active;
  }

  &--moving {
    opacity: 0.5;
    pointer-events: none;
    z-index: $--xd-zindex-element-moving !important;
  }

  &:has([data-status='true']) {
    z-index: $--xd-zindex-element-moving;
  }

  &__outline {
    @include theme('blue-1', outline, '1.5px solid');

    pointer-events: none;
    position: absolute;
  }

  &__events-area {
    @include enter-delay(100ms);
  }
}
