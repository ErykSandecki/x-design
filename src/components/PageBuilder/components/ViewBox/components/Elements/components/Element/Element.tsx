import { isNumber } from 'lodash';
import { FC, memo, ReactNode, useRef } from 'react';
import { useSelector } from 'react-redux';

// components
import EventsArea from './components/EventsArea/EventsArea';
import Outline from './components/Outline/Outline';
import { Box } from 'shared';

// core
import { useRefs } from 'pages/PageBuilderPage/core/RefsProvider';

// hooks
import { useElementEvents } from './hooks/useElementEvents';
import { useTheme } from 'hooks';

// others
import {
  className as classNameElement,
  classNames,
  classes,
} from './classNames';
import { DATA_STATUS_ATTRIBUTE } from './constants';

// store
import {
  elementDynamicDataSelectorCreator,
  eventSelectorCreator,
  isDraggableSelectorCreator,
  isHoverSelectorCreator,
  isSelectedElementSelectorCreator,
  multipleSelectedElementsSelector,
  counterAngleSelectorCreator,
} from 'store/pageBuilder/selectors';

// styles
import styles from './element.scss';

// types
import { ElementType, TColor, TElement } from 'types';
import { MouseMode } from 'types/enums/mouseMode';

// utils
import { getAbsolutePosition } from '../../../../utils/getAbsolutePosition';
import { getPosition } from './utils/getPosition';

type TElementProps = {
  classes: typeof classes;
  children: (
    coordinates: TElement['coordinates'],
    height: TElement['height'],
    hover: boolean,
    rotate: TElement['rotate'],
    selected: boolean,
    width: TElement['width'],
  ) => ReactNode;
  id: TElement['id'];
  mouseMode: MouseMode;
  parentId: TElement['parentId'];
  type: ElementType;
};

const Element: FC<TElementProps> = ({
  classes,
  children,
  id,
  mouseMode,
  parentId,
  type,
}) => {
  const counterAngle = useSelector(counterAngleSelectorCreator(parentId));
  const isDraggable = useSelector(isDraggableSelectorCreator(id));
  const isHover = useSelector(isHoverSelectorCreator(id));
  const isSelected = useSelector(isSelectedElementSelectorCreator(id));
  const isMultiple = useSelector(multipleSelectedElementsSelector);
  const elementRef = useRef<HTMLDivElement>(null);
  const elementDynamicData = useSelector(elementDynamicDataSelectorCreator(id));
  const { itemsRefs, zoomContentRef } = useRefs();
  const { alignment, coordinates } = elementDynamicData;
  const { background, height, position, rotate, width } = elementDynamicData;
  const { x, y } = coordinates;
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);
  const { ...events } = useElementEvents(
    coordinates,
    elementRef,
    id,
    isMultiple,
    isSelected,
    mouseMode,
    parentId,
    position,
    type,
  );
  const isMultipleMoving = useSelector(
    eventSelectorCreator('isMultipleMoving'),
  ) as boolean;
  const isFocused = isHover || isSelected;
  const displayEventsArea = !isDraggable && !isMultiple && isSelected;
  const displayOutline = isFocused;
  const isMoving = isDraggable || (isMultipleMoving && isSelected);
  const { x1, y1 } = getAbsolutePosition(id, itemsRefs, zoomContentRef);

  return (
    <Box
      classes={{
        className: cx(
          classes.className,
          classNamesWithTheme[classNameElement].name,
          [classNamesWithTheme[classNameElement].modificators.moving, isMoving],
        ),
      }}
      id={id}
      ref={elementRef}
      style={{
        ...getPosition(alignment, counterAngle, x, y),
        backgroundColor: 'unset',
        height: isNumber(height) ? `${height}px` : height,
        position,
        width: isNumber(width) ? `${width}px` : width,
      }}
    >
      <Box
        attributes={{ [DATA_STATUS_ATTRIBUTE]: isSelected ? 'true' : 'false' }}
        classes={{
          className: cx(classes.className, classNamesWithTheme.wrapper.name, [
            classNamesWithTheme.wrapper.modificators.hover,
            isHover,
          ]),
        }}
        style={{
          backgroundColor: (background.properties as TColor).color,
          height: '100%',
          transform: `rotate(${rotate - counterAngle}deg)`,
          width: '100%',
        }}
        {...events}
      >
        {children(coordinates, height, isHover, rotate, isSelected, width)}
        {displayOutline && (
          <Outline
            height={height}
            rotate={rotate - counterAngle}
            x={x1}
            y={y1}
            width={width}
          />
        )}
        {displayEventsArea && (
          <EventsArea
            absoluteCoordinates={{ x: x1, y: y1 }}
            counterAngle={counterAngle}
            elementRef={elementRef}
            height={height}
            id={id}
            mouseMode={mouseMode}
            relativeCoordinates={{ x, y }}
            rotate={rotate}
            width={width}
          />
        )}
      </Box>
    </Box>
  );
};

export default memo(Element);
