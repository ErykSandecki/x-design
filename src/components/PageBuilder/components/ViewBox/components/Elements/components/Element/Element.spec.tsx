import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

// components
import Element from './Element';

// core
import { RefsProvider } from 'pages/PageBuilderPage/core/RefsProvider';

// mocks
import {
  elementAllDataMock,
  elementDynamicDataMock,
  elementStaticDataMock,
  eventsMock,
  pageBuilderStateMock,
  selectedElementMock,
} from 'test/mocks/reducer/pageBuilderMock';

// others
import { REDUCER_KEY as PAGE_BUILDER } from 'store/pageBuilder/actionsType';

// store
import { configureStore } from 'store/store';

// types
import { ElementType } from 'types';
import { MouseMode } from 'types/enums/mouseMode';

const element = document.createElement('div');
const overlayContainer = document.createElement('div');

const sharedRefs = {
  [elementAllDataMock.id]: element,
};

const stateMock = {
  [PAGE_BUILDER]: {
    ...pageBuilderStateMock[PAGE_BUILDER],
    pages: {
      ...pageBuilderStateMock[PAGE_BUILDER].pages,
      ['0']: {
        ...pageBuilderStateMock[PAGE_BUILDER].pages['0'],
        elements: {
          ...pageBuilderStateMock[PAGE_BUILDER].pages['0'].elements,
          dynamicData: {
            ...pageBuilderStateMock[PAGE_BUILDER].pages['0'].elements
              .dynamicData,
            [elementDynamicDataMock.id]: elementDynamicDataMock,
          },
          staticData: {
            ...pageBuilderStateMock[PAGE_BUILDER].pages['0'].elements
              .staticData,
            [elementStaticDataMock.id]: elementStaticDataMock,
          },
        },
      },
    },
  },
};

jest.mock('lodash', () => ({
  ...(jest.requireActual('lodash') as object),
  defer: (callback: any) => callback(),
}));

describe('Element snapshots', () => {
  beforeAll(() => {
    element.style.height = '100px';
    element.style.width = '100px';
    document.body.appendChild(overlayContainer);
  });

  it('should render Element', () => {
    // mock
    const store = configureStore(stateMock);

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <RefsProvider
          itemsRefs={sharedRefs}
          overlayContainerRefHtml={overlayContainer}
        >
          <Element
            classes={{ className: 'className' }}
            id={selectedElementMock.id}
            mouseMode={MouseMode.default}
            parentId={selectedElementMock.parentId}
            type={ElementType.frame}
          >
            {() => <></>}
          </Element>
        </RefsProvider>
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render Element when is selected', () => {
    // mock
    const store = configureStore({
      ...stateMock,
      [PAGE_BUILDER]: {
        ...stateMock[PAGE_BUILDER],
        pages: {
          ...stateMock[PAGE_BUILDER].pages,
          ['0']: {
            ...stateMock[PAGE_BUILDER].pages['0'],
            selectedElements: [selectedElementMock],
          },
        },
      },
    });

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <RefsProvider
          itemsRefs={sharedRefs}
          overlayContainerRefHtml={overlayContainer}
        >
          <Element
            classes={{ className: 'className' }}
            id={selectedElementMock.id}
            mouseMode={MouseMode.default}
            parentId={selectedElementMock.parentId}
            type={ElementType.frame}
          >
            {() => <></>}
          </Element>
        </RefsProvider>
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render Element when width & height is auto', () => {
    // mock
    window.getComputedStyle = () =>
      ({ height: '100px', width: '100px' }) as CSSStyleDeclaration;

    // mock
    const store = configureStore({
      ...stateMock,
      [PAGE_BUILDER]: {
        ...stateMock[PAGE_BUILDER],
        pages: {
          ...stateMock[PAGE_BUILDER].pages,
          ['0']: {
            ...stateMock[PAGE_BUILDER].pages['0'],
            elements: {
              ...stateMock[PAGE_BUILDER].pages['0'].elements,
              dynamicData: {
                ...stateMock[PAGE_BUILDER].pages['0'].elements.dynamicData,
                [elementDynamicDataMock.id]: {
                  ...elementDynamicDataMock,
                  height: 'auto',
                  width: 'auto',
                },
              },
              staticData: {
                ...stateMock[PAGE_BUILDER].pages['0'].elements.staticData,
                [elementStaticDataMock.id]: elementStaticDataMock,
              },
            },
          },
        },
      },
    });

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <RefsProvider
          itemsRefs={sharedRefs}
          overlayContainerRefHtml={overlayContainer}
        >
          <Element
            classes={{ className: 'className' }}
            id={selectedElementMock.id}
            mouseMode={MouseMode.default}
            parentId={selectedElementMock.parentId}
            type={ElementType.frame}
          >
            {() => <></>}
          </Element>
        </RefsProvider>
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render Element when is selected & is multiple moving', () => {
    // mock
    const store = configureStore({
      ...stateMock,
      [PAGE_BUILDER]: {
        ...stateMock[PAGE_BUILDER],
        events: {
          ...eventsMock,
          isMultipleMoving: true,
        },
        pages: {
          ...stateMock[PAGE_BUILDER].pages,
          ['0']: {
            ...stateMock[PAGE_BUILDER].pages['0'],
            selectedElements: [selectedElementMock],
          },
        },
      },
    });

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <RefsProvider
          itemsRefs={sharedRefs}
          overlayContainerRefHtml={overlayContainer}
        >
          <Element
            classes={{ className: 'className' }}
            id={selectedElementMock.id}
            mouseMode={MouseMode.default}
            parentId={selectedElementMock.parentId}
            type={ElementType.frame}
          >
            {() => <></>}
          </Element>
        </RefsProvider>
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
