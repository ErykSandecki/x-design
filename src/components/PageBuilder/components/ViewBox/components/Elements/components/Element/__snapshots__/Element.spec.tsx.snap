// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Element snapshots should render Element 1`] = `
<DocumentFragment>
  <div
    class="css-wcnq0t className Element Element--light"
    data-test-box=""
    id="1"
    style="left: 0px; top: 0px; transform: rotate(0deg) translate(0, 0); transform-origin: center center; height: 100px; position: absolute; width: 100px;"
  >
    <div
      class="css-wcnq0t className Element__wrapper Element__wrapper--light"
      data-status="false"
      data-test-box=""
      style="background-color: rgb(255, 255, 255); height: 100%; transform: rotate(0deg); width: 100%;"
    />
  </div>
</DocumentFragment>
`;

exports[`Element snapshots should render Element when is selected & is multiple moving 1`] = `
<DocumentFragment>
  <div
    class="css-wcnq0t className Element Element--light Element--moving Element--moving--light"
    data-test-box=""
    id="1"
    style="left: 0px; top: 0px; transform: rotate(0deg) translate(0, 0); transform-origin: center center; height: 100px; position: absolute; width: 100px;"
  >
    <div
      class="css-wcnq0t className Element__wrapper Element__wrapper--light"
      data-status="true"
      data-test-box=""
      style="background-color: rgb(255, 255, 255); height: 100%; transform: rotate(0deg); width: 100%;"
    >
      <div
        class="css-wcnq0t Outline Outline--light"
        data-test-box=""
        style="height: 100px; left: 0px; top: 0px; transform: rotate(0deg); width: 100px;"
      />
      <div
        class="css-u1i9af"
        data-test-box=""
        style="height: 100px; left: 0px; top: 0px; transform: rotate(0deg); transform-origin: center center; width: 100px;"
      >
        <div
          class="css-wcnq0t"
          data-test-box=""
          style="height: 100px; width: 100px;"
        />
        <svg
          class="Corners Corners--light"
          style="height: 102px; left: -1px; top: -1px; width: 102px;"
        >
          <rect
            height="6"
            rx="1.5"
            width="6"
            x="-3"
            y="-3"
          />
          <rect
            height="6"
            rx="1.5"
            width="6"
            x="-3"
            y="97.5"
          />
          <rect
            height="6"
            rx="1.5"
            width="6"
            x="97.5"
            y="-3"
          />
          <rect
            height="6"
            rx="1.5"
            width="6"
            x="97.5"
            y="97.5"
          />
        </svg>
        <div
          class="TransformArea TransformArea--light"
          style="height: 100px; width: 100px;"
        >
          <div
            class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--east TransformArea__anchor-resize--east--light"
            data-test-anchor-resize="east"
          />
          <div
            class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--north TransformArea__anchor-resize--north--light"
            data-test-anchor-resize="north"
          />
          <div
            class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--northEast TransformArea__anchor-resize--northEast--light"
            data-test-anchor-resize="north-east"
          />
          <div
            class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--northWest TransformArea__anchor-resize--northWest--light"
            data-test-anchor-resize="north-west"
          />
          <div
            class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--south TransformArea__anchor-resize--south--light"
            data-test-anchor-resize="south"
          />
          <div
            class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--southEast TransformArea__anchor-resize--southEast--light"
            data-test-anchor-resize="south-east"
          />
          <div
            class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--southWest TransformArea__anchor-resize--southWest--light"
            data-test-anchor-resize="south-west"
          />
          <div
            class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--west TransformArea__anchor-resize--west--light"
            data-test-anchor-resize="west"
          />
          <div
            class="TransformArea__anchor-rotate TransformArea__anchor-rotate--light TransformArea__anchor-rotate--northEast TransformArea__anchor-rotate--northEast--light"
            data-test-anchor-rotate="north-east"
          />
          <div
            class="TransformArea__anchor-rotate TransformArea__anchor-rotate--light TransformArea__anchor-rotate--northWest TransformArea__anchor-rotate--northWest--light"
            data-test-anchor-rotate="north-west"
          />
          <div
            class="TransformArea__anchor-rotate TransformArea__anchor-rotate--light TransformArea__anchor-rotate--southEast TransformArea__anchor-rotate--southEast--light"
            data-test-anchor-rotate="south-east"
          />
          <div
            class="TransformArea__anchor-rotate TransformArea__anchor-rotate--light TransformArea__anchor-rotate--southWest TransformArea__anchor-rotate--southWest--light"
            data-test-anchor-rotate="south-west"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Element snapshots should render Element when is selected 1`] = `
<DocumentFragment>
  <div
    class="css-wcnq0t className Element Element--light"
    data-test-box=""
    id="1"
    style="left: 0px; top: 0px; transform: rotate(0deg) translate(0, 0); transform-origin: center center; height: 100px; position: absolute; width: 100px;"
  >
    <div
      class="css-wcnq0t className Element__wrapper Element__wrapper--light"
      data-status="true"
      data-test-box=""
      style="background-color: rgb(255, 255, 255); height: 100%; transform: rotate(0deg); width: 100%;"
    >
      <div
        class="css-wcnq0t Outline Outline--light"
        data-test-box=""
        style="height: 100px; left: 0px; top: 0px; transform: rotate(0deg); width: 100px;"
      />
      <div
        class="css-u1i9af"
        data-test-box=""
        style="height: 100px; left: 0px; top: 0px; transform: rotate(0deg); transform-origin: center center; width: 100px;"
      >
        <div
          class="css-wcnq0t"
          data-test-box=""
          style="height: 100px; width: 100px;"
        />
        <svg
          class="Corners Corners--light"
          style="height: 102px; left: -1px; top: -1px; width: 102px;"
        >
          <rect
            height="6"
            rx="1.5"
            width="6"
            x="-3"
            y="-3"
          />
          <rect
            height="6"
            rx="1.5"
            width="6"
            x="-3"
            y="97.5"
          />
          <rect
            height="6"
            rx="1.5"
            width="6"
            x="97.5"
            y="-3"
          />
          <rect
            height="6"
            rx="1.5"
            width="6"
            x="97.5"
            y="97.5"
          />
        </svg>
        <div
          class="TransformArea TransformArea--light"
          style="height: 100px; width: 100px;"
        >
          <div
            class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--east TransformArea__anchor-resize--east--light"
            data-test-anchor-resize="east"
          />
          <div
            class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--north TransformArea__anchor-resize--north--light"
            data-test-anchor-resize="north"
          />
          <div
            class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--northEast TransformArea__anchor-resize--northEast--light"
            data-test-anchor-resize="north-east"
          />
          <div
            class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--northWest TransformArea__anchor-resize--northWest--light"
            data-test-anchor-resize="north-west"
          />
          <div
            class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--south TransformArea__anchor-resize--south--light"
            data-test-anchor-resize="south"
          />
          <div
            class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--southEast TransformArea__anchor-resize--southEast--light"
            data-test-anchor-resize="south-east"
          />
          <div
            class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--southWest TransformArea__anchor-resize--southWest--light"
            data-test-anchor-resize="south-west"
          />
          <div
            class="TransformArea__anchor-resize TransformArea__anchor-resize--light TransformArea__anchor-resize--west TransformArea__anchor-resize--west--light"
            data-test-anchor-resize="west"
          />
          <div
            class="TransformArea__anchor-rotate TransformArea__anchor-rotate--light TransformArea__anchor-rotate--northEast TransformArea__anchor-rotate--northEast--light"
            data-test-anchor-rotate="north-east"
          />
          <div
            class="TransformArea__anchor-rotate TransformArea__anchor-rotate--light TransformArea__anchor-rotate--northWest TransformArea__anchor-rotate--northWest--light"
            data-test-anchor-rotate="north-west"
          />
          <div
            class="TransformArea__anchor-rotate TransformArea__anchor-rotate--light TransformArea__anchor-rotate--southEast TransformArea__anchor-rotate--southEast--light"
            data-test-anchor-rotate="south-east"
          />
          <div
            class="TransformArea__anchor-rotate TransformArea__anchor-rotate--light TransformArea__anchor-rotate--southWest TransformArea__anchor-rotate--southWest--light"
            data-test-anchor-rotate="south-west"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Element snapshots should render Element when width & height is auto 1`] = `
<DocumentFragment>
  <div
    class="css-wcnq0t className Element Element--light"
    data-test-box=""
    id="1"
    style="left: 0px; top: 0px; transform: rotate(0deg) translate(0, 0); transform-origin: center center; height: auto; position: absolute; width: auto;"
  >
    <div
      class="css-wcnq0t className Element__wrapper Element__wrapper--light"
      data-status="false"
      data-test-box=""
      style="background-color: rgb(255, 255, 255); height: 100%; transform: rotate(0deg); width: 100%;"
    />
  </div>
</DocumentFragment>
`;
